2011-12-07  Nicolas Setton  <setton@adacore.com>

	* glade-gtk.c (glade_gtk_cell_renderer_sync_attributes): The widget
	could be null here, for instance when the model was associated to
	a GtkCombo with entry.

	(Not submitted: fixed in head by calling glade_widget_get_parent).

2011-01-20  Nicolas Setton  <setton@adacore.com>

	* glade-catalog.c (glade_catalog_load_all): Disable message listing
	the missing icons.

2011-01-20  Nicolas Setton  <setton@adacore.com>

	* glade-utils.c (glade_util_get_file_mtime): Do not attempt to run g_stat
	under Windows, to prevent segvs.

2011-01-04  Nicolas Setton  <setton@adacore.com>

	* glade-widget-adaptor.c
	(gwa_displayable_values_check): Disable harmless messages.

2011-01-03  Nicolas Setton  <setton@adacore.com>

	* glade-app.c (build_package_paths): Improve relocatability, by allowing
	the user to set the prefix for icons through the GLADE_BASE_DIR environment
	variable.

*** gladeui/glade-app.c.orig	2011-01-20 13:02:59.000000000 +0100
--- gladeui/glade-app.c	2011-01-20 12:55:41.000000000 +0100
***************
*** 42,47 ****
--- 42,49 ----
  #include "glade-accumulators.h"
  
  #include <string.h>
+ #include <stdlib.h>
+ 
  #include <glib.h>
  #include <glib/gstdio.h>
  #include <glib/gi18n-lib.h>
*************** build_package_paths (void)
*** 393,404 ****
  
  	g_free (prefix);
  #else
! 	catalogs_dir = g_strdup (GLADE_CATALOGSDIR);
! 	modules_dir  = g_strdup (GLADE_MODULESDIR);
! 	plugins_dir  = g_strdup (GLADE_PLUGINSDIR);
! 	pixmaps_dir  = g_strdup (GLADE_PIXMAPSDIR);
! 	locale_dir   = g_strdup (GLADE_LOCALEDIR);
! 	bin_dir      = g_strdup (GLADE_BINDIR);
  #endif
  }
  
--- 395,421 ----
  
  	g_free (prefix);
  #else
!        gchar *prefix;
! 
!        prefix = getenv("GLADE_BASE_DIR");
! 
!        if (prefix)
!                {
!                        pixmaps_dir  = g_build_filename (prefix, "share", PACKAGE, "pixmaps", NULL);
!                        catalogs_dir = g_build_filename (prefix, "share", PACKAGE, "catalogs", NULL);
!                        modules_dir  = g_build_filename (prefix, "lib", PACKAGE, "modules", NULL);
!                        locale_dir   = g_build_filename (prefix, "share", "locale", NULL);
!                        bin_dir      = g_build_filename (prefix, "bin", NULL);
!                }
!        else
!                {
!                        catalogs_dir = g_strdup (GLADE_CATALOGSDIR);
!                        modules_dir  = g_strdup (GLADE_MODULESDIR);
!                        plugins_dir  = g_strdup (GLADE_PLUGINSDIR);
!                        pixmaps_dir  = g_strdup (GLADE_PIXMAPSDIR);
!                        locale_dir   = g_strdup (GLADE_LOCALEDIR);
!                }
! 
  #endif
  }
  
*** gladeui/glade-widget-adaptor.c.orig	2011-01-20 13:04:07.000000000 +0100
--- gladeui/glade-widget-adaptor.c	2011-01-20 12:51:32.000000000 +0100
*************** gwa_displayable_values_check (GladeWidge
*** 2044,2049 ****
--- 2044,2050 ----
  		    klass->pspec->value_type != GLADE_TYPE_STOCK_IMAGE)
  		{
  			/* We do not need displayable values if the property is not visible */
+                         if (0)
  			g_message ("No displayable values for %sproperty %s::%s", 
  				   (packing) ? "child " : "",
  				   adaptor->name, klass->id);
*** gladeui/glade-utils.c.orig	2011-01-20 13:05:38.000000000 +0100
--- gladeui/glade-utils.c	2011-01-20 13:08:14.000000000 +0100
*************** glade_util_get_file_mtime
*** 1860,1865 ****
--- 1860,1869 ----
  {
  	struct stat info;
  	gint retval;
+ 
+ #ifdef G_OS_WIN32
+ 	return (time_t) 0;
+ #endif
  	
  	retval = g_stat (filename, &info);
  	
*** gladeui/glade-catalog.c.orig	2011-01-20 16:52:15.000000000 +0100
--- gladeui/glade-catalog.c	2011-01-20 16:55:59.000000000 +0100
*************** glade_catalog_load_all (void)
*** 574,579 ****
--- 574,580 ----
  		catalog_load (catalog);
  	}
  
+ #if 0
  	/* Print a summery of widget adaptors missing icons.
  	 */
  	adaptors = glade_widget_adaptor_list_adaptors ();
*************** glade_catalog_load_all (void)
*** 594,599 ****
--- 595,601 ----
  
         	}
  	g_list_free (adaptors);
+ #endif
  
  	if (icon_warning)
  	{
--- plugins/gtk+/glade-gtk.c.orig	2011-12-07 15:08:53.000000000 +0100
+++ plugins/gtk+/glade-gtk.c	2011-12-07 15:09:21.000000000 +0100
@@ -10644,7 +10644,7 @@ glade_gtk_cell_renderer_sync_attributes 
 	/* Apply attributes to renderer when bound to a model in runtime */
 	widget = glade_widget_get_from_gobject (object);
 		
-	if (widget->parent == NULL) return FALSE;
+	if ((widget == NULL) || (widget->parent == NULL)) return FALSE;
 
 	/* When creating widgets, sometimes the parent is set before parenting happens,
 	 * here we have to be careful for that..
